provider/src/iosMain/kotlin/at/asitplus/crypto/provider/os/IosKeychainProvider.kt

```diff
@@ -0,0 +1,420 @@
+                throw NoSuchElementException("Key with alias $alias already exists")
+        }
+        deleteSigningKey(alias) /* make sure there are no leftover private keys */
+
+        val config = DSL.resolve(::iosSigningKeyConfiguration, configure)
+
+        val publicKeyBytes: ByteArray = memScoped {
+            val availability = config.hardware.v.let { c-> when (c.availability) {
+                iosSecureEnclaveConfiguration.Availability.ALWAYS -> if (c.allowBackup) kSecAttrAccessibleAlways else kSecAttrAccessibleAlwaysThisDeviceOnly
+                iosSecureEnclaveConfiguration.Availability.AFTER_FIRST_UNLOCK -> if (c.allowBackup) kSecAttrAccessibleAfterFirstUnlock else kSecAttrAccessibleAfterFirstUnlockThisDeviceOnly
+                iosSecureEnclaveConfiguration.Availability.WHILE_UNLOCKED -> if (c.allowBackup) kSecAttrAccessibleWhenUnlocked else kSecAttrAccessibleWhenUnlockedThisDeviceOnly
+            } }
+            val attr = createCFDictionary {
+                when (val alg = config._algSpecific.v) {
+                    is SigningKeyConfiguration.ECConfiguration -> {
+                        kSecAttrKeyType mapsTo kSecAttrKeyTypeEC
+                        kSecAttrKeySizeInBits mapsTo alg.curve.coordinateLength.bits.toInt()
+                    }
+                    is SigningKeyConfiguration.RSAConfiguration -> {
+                        kSecAttrKeyType mapsTo kSecAttrKeyTypeRSA
+                        kSecAttrKeySizeInBits mapsTo alg.bits
+                    }
+                }
+                val useSecureEnclave = when (config.hardware.v.backing) {
+                    is REQUIRED -> true
+                    is PREFERRED -> isSecureEnclaveSupportedCurve(config._algSpecific.v)
+                    is DISCOURAGED -> false
+                }
+                if (useSecureEnclave) {
+                    kSecAttrTokenID mapsTo kSecAttrTokenIDSecureEnclave
+                }
+                kSecPrivateKeyAttrs mapsTo createCFDictionary {
+                    kSecAttrLabel mapsTo alias
+                    kSecAttrIsPermanent mapsTo true
+                    kSecAttrApplicationTag mapsTo KeychainTags.PRIVATE_KEYS
+                    when (val factors = config.hardware.v.protection.v?.factors?.v) {
+                        null -> {
+                            kSecAttrAccessible mapsTo availability
+                        }
+                        else -> {
+                            kSecAttrAccessControl mapsTo corecall {
+                                SecAccessControlCreateWithFlags(
+                                    null, availability,
+                                    when {
+                                        (factors.biometry && factors.deviceLock) -> kSecAccessControlUserPresence
+                                        factors.biometry -> kSecAccessControlBiometryAny
+                                        else -> kSecAccessControlDevicePasscode
```

Comment:
I'm pretty sure we want to be able to fully, manually specify this ba accessing those constants and not through a wrapper. but do present your arguments and I may be convinced otherwise